composer/trainer/trainer.py

```diff
@@ -1158,6 +1164,36 @@ def __init__(
         # Run Event.INIT
         self.engine.run_event(Event.INIT)
 
+        # If the experiment is being tracked with an `MLFlowLogger`, then MLFlow experiment and run are available
+        # after Event.INIT.
+        if save_folder is not None:
+            mlflow_logger = None
+            for destination in self.logger.destinations:
+                if isinstance(destination, MLFlowLogger):
+                    mlflow_logger = destination
+                    break
+
+            if mlflow_logger is not None:
+                mlflow_experiment_id = mlflow_logger._experiment_id
+                mlflow_run_id = mlflow_logger._run_id
+
+                # Log the experiment and run IDs as hyperparameters.
+                self.logger.log_hyperparameters({
+                    'mlflow_experiment_id': mlflow_experiment_id,
+                    'mlflow_run_id': mlflow_run_id
+                })
```

Comment:
This should just be done in side the MLFlow logger (which is a callback) on Event.init

Reply:
If there are multiple loggers, wouldn't we want to wait until post INIT before logging hyperparameters to make sure all loggers are initialized? If we did it inside `MLFlowLogger.init()`, then e.g., the `MosaicMLLogger` might not be initialized, right?

Reply:
hm... I see. Can you do `Event.AFTER_LOAD`? Composer pulling out specific callbacks and having them operate is a big anti-pattern

Reply:
I think we can do the hyperparameter logging in `Event.AFTER_LOAD`. However, see my answer below for why we need to update the `save_folder` with the MLflow run information, so we still need some way to pull that info out inside `Trainer`

Reply:
Let's still move this log line though?